src/platform/w3dengine/client/w3ddisplay.h

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file
+ *
+ * @author Duncans_pumpkin
+ *
+ * @brief W3D class for the display handling.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+
+#include "coord.h"
+#include "display.h"
+#include "light.h"
+#include "render2d.h"
+#include "scene.h"
+
+class W3DAssetManager;
+
+class W3DDisplay final : public Display
+{
+public:
+    W3DDisplay();
+    ~W3DDisplay();
+    virtual void Init() override;
+    virtual void Reset() override;
+    virtual void Draw() override;
+    virtual void Set_Width(uint32_t width) override;
+    virtual void Set_Height(uint32_t height) override;
+    virtual bool Set_Display_Mode(uint32_t width, uint32_t height, uint32_t bits, bool windowed) override;
+    virtual int Get_Display_Mode_Count() override;
+    virtual void Get_Display_Mode_Description(int id, int *width, int *height, int *bit_depth) override;
+    virtual void Do_Smart_Asset_Purge_And_Preload(const char *) override;
+    virtual VideoBuffer *Create_VideoBuffer() override;
+    virtual void Set_Clip_Region(IRegion2D *region) override;
+    virtual bool Is_Clipping_Enabled() override { return m_isClippedEnabled; }
+    virtual void Enable_Clipping(bool isEnabled) override { m_isClippedEnabled = isEnabled; }
+    virtual void Set_Time_Of_Day(TimeOfDayType time) override;
+    virtual void Create_Light_Pulse(const Coord3D *, const RGBColor *, float, float, unsigned, unsigned) override;
+    virtual void Draw_Line(int32_t x1, int32_t y1, int32_t x2, int32_t y2, float width, uint32_t colour) override;
```

Comment:
check if order is right in compiled binary, 8 args first ,7 args after

Reply:
Okay I've swapped that now in Display and W3DDisplay